skycoin/skywire
# add skywire-cli subcommands

Did you run `make format && make check`?
yes

~~Fixes pull request review issue from skycoin/skybian#134 ; creating http endpoint for querying the hypervisor's public key~~
*re-implemented srvpk into skybian much more simply [here](https://github.com/the-skycoin-project/skybian/blob/rework/script/srvpk.service)



 Changes:	
- ~~Creates HTTP endpoint  for `skywire cli visor pk` with `skywire-cli hv srvpk`~~
- `skywire-cli hv` subcommands (see systray PR)
- Minor restructuring of `skywire-cli` visor commands
- fix api endpoints must return errors for remotevisors and vpnserverslist

How to test this PR:

Clone my fork, checkout srvpk branch and start a visor
```
git clone https://github.com/the-skycoin-project/skywire
cd skywire
git checkout srvpk
make run-source
```

~~start the http endpoint in a new terminal or tab~~
~~(`ctrl`+`shift`+`t`)~~
~~```~~
~~go run cmd/skywire-cli/skywire-cli.go hv srvpk~~
~~```~~

~~query the http endpoint at [http://127.0.0.1:7998](http://127.0.0.1:7998) in the browser~~
~~or in a new terminal or tab using curl~~
~~(`ctrl`+`shift`+`t`)~~
~~```~~
~~curl http://127.0.0.1:7998~~
~~```~~

~~The output should be the same as the following:~~
~~```~~
~~go run cmd/skywire-cli/skywire-cli.go visor pk~~
~~```~~
~~without the trailing newline~~
